src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -612,7 +612,10 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         const auto profileViewModel{ _viewModelForProfile(newProfile, _settingsClone) };
         profileViewModel.SetupAppearances(_colorSchemesPageVM.AllColorSchemes());
         const auto navItem{ _CreateProfileNavViewItem(profileViewModel) };
-        SettingsNav().MenuItems().InsertAt(index, navItem);
+
+        auto navItems{ SettingsNav().MenuItemsSource().try_as<Collections::IVector<IInspectable>>() };
+        navItems.InsertAt(index, navItem);
+        SettingsNav().MenuItemsSource(navItems);

```

Comment:
what's the purpose of making it observable if we're also resetting it? this mixes the metaphors...